discord/ui/item.py

```diff
@@ -104,22 +119,59 @@ def row(self) -> Optional[int]:
 
     @row.setter
     def row(self, value: Optional[int]) -> None:
+        if self._is_v2():
+            # row is ignored on v2 components
+            return
+
         if value is None:
             self._row = None
         elif 5 > value >= 0:
             self._row = value
         else:
             raise ValueError('row cannot be negative or greater than or equal to 5')
 
+        if self._rendered_row is None:
+            self._rendered_row = value
+
     @property
     def width(self) -> int:
         return 1
 
     @property
     def view(self) -> Optional[V]:
-        """Optional[:class:`View`]: The underlying view for this item."""
+        """Optional[Union[:class:`View`, :class:`LayoutView`]]: The underlying view for this item."""
         return self._view
 
+    @property
+    def id(self) -> Optional[int]:
+        """Optional[:class:`int`]: The ID of this component."""
+        return self._id
+
+    @id.setter
+    def id(self, value: Optional[int]) -> None:
+        self._id = value
+
+    @property
+    def parent(self) -> Optional[Item[V]]:
+        """Optional[:class:`Item`]: This item's parent. If this item is a :class:`View` this will
+        always be ``None``.
```

Comment:
```suggestion
        """Optional[:class:`Item`]: This item's parent. Only components that can have children
        can be parents. Any item that has :class:`View` as a view will have this set to `None`
        since only :class:`LayoutView` component v2 items can contain "container" like items.
        
        .. versionadded:: 2.6
```